app/models/workers_scaling/experiment_resources_interface.rb

```diff
@@ -0,0 +1,111 @@
+require 'infrastructure_facades/infrastructure_facade_factory'
+require 'infrastructure_facades/infrastructure_errors'
+include InfrastructureErrors
+##
+# Experiment interface to schedule and maintain computational resources
+module WorkersScaling
+  class ExperimentResourcesInterface
+
+    def initialize(experiment_id, user_id)
+      # TODO list if disallowed infrastructures
+      @experiment_id = experiment_id.to_s
+      @user_id = BSON::ObjectId(user_id.to_s)
+      @facades_cache = {}
+    end
+
+    ##
+    # Schedules given amount of workers onto infrastructure and returns theirs sm_uuids
+    # In case of error returns nil
+    # Additional params:
+    # * time_limit
+    # * proxy
+    # Raises InfrastructureError
+    def schedule_workers(amount, infrastructure_name, params = {})
+      begin
+        #TODO to_s, to_sym
+        params['time_limit'] = 60 if params['time_limit'].nil?
+        params.merge! onsite_monitoring: true
+        get_facade_for(infrastructure_name)
+          .start_simulation_managers(@user_id, amount, @experiment_id, params)
+          .map &:sm_uuid
+      rescue InvalidCredentialsError, NoCredentialsError
+        # TODO inform user about credentials error
+        raise
+      end
+    end
+
+    ##
+    # Returns list of available infrastructures for experiment
+    def get_available_infrastructures
+      InfrastructureFacadeFactory.list_infrastructures(@user_id)
+          .flat_map {|x| x.has_key?(:children) ? x[:children] : x }
+          .select {|x| x[:enabled]}
+          .map {|x| x[:infrastructure_name].to_sym}
+    end
+
+    ##
+    # Returns statistics about available infrastructure:
+    #   * workers_count
+    # params:
+    #   * infrastructure_names
+    # Raises InfrastructureError
+    def get_infrastructures_statistics(params = {})
+      infrastructure_names = params[:infrastructure_names] || get_available_infrastructures
+      infrastructure_names.map {|name| [name, get_infrastructure_statistics(name, params)]}.to_h
+    end
+
+    ##
+    # Returns statistics about given infrastructure, statistics description in #get_infrastructures_statistics
```

Comment:
j.w. napisać, że to jest Hash (chyba, że w get_infrastructures.. będzie to napisane)